src/Fabulous.Avalonia/Views/Controls/AutoCompleteBox.fs

```diff
@@ -80,6 +80,24 @@ module AutoCompleteBox =
 
                     target.Loaded.AddHandler(bindAndCleanUp))
 
+    /// Allows setting the ItemTemplate on an AutoCompleteBox
+    let ItemTemplate =
+        Attributes.defineSimpleScalar<obj -> Widget>
```

Comment:
Is this the correct `Attributes.define*` variant to use in this case? I used it (without fully understanding what its use case is) because I was looking for something that gives me access to a `ViewNode` to pass on the `WidgetDataTemplate` below.

Reply:
Yes this is the correct `define` function to be used here. For convenience and to be able to use our `ScalarAttributeComparers.equalityCompare` helpers we use `WidgetItems` that already contains a `Template: obj -> Widget` but also the `OriginalItems` that we use for the comparison.

Reply:
I saw the `WidgetItems` abstraction used for other item controls that allow templating. I chose not to use it because the `AutoCompleteBox` is special in the sense that it supports the populator as an async item source. So I was looking for a solution that can be used independent of the choice of item source.

Reply:
Let me investigate a little more abut how we can achieve this.

Reply:
Ok. I had a more closer look and I think your implementation is Ok. As you mentioned the item `itemTemplate` is better as a modifier and independent from the item source.